context/context.go

```diff
@@ -202,3 +256,60 @@ func (c *ctx) returnToPool() {
 	c.Reset()
 	c.pool.Put(c)
 }
+
+func (c *ctx) NewChildContext() Context {
+	var child Context
+	if c.pool == nil {
+		child = NewContext()
+	} else {
+		child = c.pool.Get()
+	}
+
+	child.SetParentCtx(c)
+
+	return child
+}
+
+func (c *ctx) SetParentCtx(parentCtx Context) {
+	// check to see if parent exists?
+	c.parent = parentCtx
+}
+
+func (c *ctx) ParentCtx() Context {
+	c.RLock()
+	defer c.RUnlock()
+	return c.parent
+}
+
+func (c *ctx) StartSpan(name string) (Context, opentracing.Span, bool) {
+	goCtx, exists := c.GoContext()
+	if !exists {
+		return c, nil, false
+	}
+
+	var (
+		sp    opentracing.Span
+		spCtx stdctx.Context
+	)
+
+	sp = opentracing.SpanFromContext(goCtx)
+	jaegerSpan, ok := sp.(*jaeger.Span)
```

Comment:
Put a note that until `opentracing` supports the `IsSampled()` method we need to cast to jaeger span.  

Maybe we can link to an open issue too? Seems there are two issues (one is a duplicate):
https://github.com/opentracing/specification/issues/8
https://github.com/opentracing/specification/issues/92

Reply:
Need to add this note above this method still FYI.